src/main/java/utils/logging/MyFileAppender.java

```diff
@@ -0,0 +1,120 @@
+package utils.logging;
+
+import org.apache.log4j.*;
+import org.apache.log4j.spi.ErrorHandler;
+import org.apache.log4j.spi.Filter;
+import org.apache.log4j.spi.LoggingEvent;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class MyFileAppender implements Appender {
+    public static final String folder = "./logging";
+    private final FileAppender fileAppender;
+
+    public MyFileAppender(){
+        fileAppender = new FileAppender();
+        HTMLLayout layout = new MyHTMLLayout();
+        layout.setTitle("All logs");
+        fileAppender.setThreshold(Level.ALL);
+        fileAppender.setFile(String.format("%s/log.html", MyFileAppender.folder));
+        fileAppender.setLayout(layout);
+        fileAppender.activateOptions();
+    }
+
+
+    @Override
+    public void addFilter(Filter filter) {
+
```

Comment:
Add `throw new NotImplementedException("This function shouldn't be called");` to all unimplemented functions.